src/components/PopoverMenu/BasePopoverMenu.js

```diff
@@ -26,13 +26,88 @@ const defaultProps = {
 };
 
 class BasePopoverMenu extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            activeMenuIndex: -1,
+        };
+        this.onModalHide = this.onModalHide.bind(this);
+    }
+
+    componentDidMount() {
+        if (!this.props.allowKeyboardNavigation) {
+            return;
+        }
+        this.setupEventHandlers();
+    }
+
+    componentWillUnmount() {
+        if (!this.props.allowKeyboardNavigation) {
+            return;
+        }
+        this.cleanupEventHandlers();
+    }
+
+    onModalHide() {
+        this.setState({activeMenuIndex: -1});
+        this.props.onMenuHide();
+    }
+
+    setupEventHandlers() {
+        if (!document) {
```

Comment:
NAB: Although this is a safe trick to manage the native platform we don't do this for other components. I leave this to other reviewer

Reply:
This was done in EmojiPicker and hence I used it here..

Reply:
on EmojiPicker this is done on the index.js file. But here this code would run on all platforms. Using `Document` global on native is bad but this check prevents the crash. But this could be an antipattern of our codebase. I am leaving this to @puneetlath .

Reply:
Because we had to maintain the `activeIndex` state, and reduce the code I put the check.

I'll update it into platform-specific code.